.circleci/meta-linter/rule_registered.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"regexp"
+	"strings"
+
+	"github.com/stoewer/go-strcase"
+)
+
+func checkRuleRegistered(aip int, name string) []error {
+	path := fmt.Sprintf("rules/aip%04d/%s.go", aip, strcase.SnakeCase(name))
+
+	// Read in the file.
+	contents, err := ioutil.ReadFile(path)
+	if err != nil {
+		return []error{err}
+	}
+
+	// Look for a rule in the file. Complain if we can not find one.
+	ruleMatch := ruleRegexp.FindStringSubmatch(string(contents))
+	if len(ruleMatch) == 0 {
+		return []error{fmt.Errorf("no rule found: %s", path)}
+	}
+
+	// Some errors are now non-fatal; start a running tab.
+	errata := []error{}
+
+	// Ensure this rule is registered within its AIP module file.
+	ruleVar := ruleMatch[1]
+	contents, err = ioutil.ReadFile(fmt.Sprintf("rules/aip%04d/aip%04d.go", aip, aip))
+	if err != nil {
+		return []error{err}
+	}
+	if !strings.Contains(string(contents), ruleVar) {
+		errata = append(errata, fmt.Errorf("rule %q for AIP-%d not registered in the AIP's AllRules", name, aip))
+	}
+
+	// Ensure that the AIP itself is registered in `rules/rules.go`.
+	contents, err = ioutil.ReadFile("rules/rules.go")
+	if err != nil {
+		errata = append(errata, err)
+		return errata
```

Comment:
I think we should return an error immediately if one is found. Otherwise, it is hard to keep tracking when we must return and when not.

Reply:
I disagree. I am going with the Golang testing philosophy -- provide all the errors we are able to at once, so people do not have to thrash.

Reply:
Okay, just a nitpicking. Thanks!